### PR TITLE
[LOOP-4289-4348] Critical alert modal and banner update

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-4289
https://tidepool.atlassian.net/browse/LOOP-4348

This update includes:
- navigating directly to settings from the modal
- modal copy update
- navigating directly to settings from the banner
- banner copy update

### CODE REVIEW

**Comment by nhamming** [comment]
@ps2 Looking for an initial review of the concept. 

Summary is that the `AlertPermissionsChecker` is now truly just a checker and reports to its delegate. The delegate (`AlertManager`) is responsible to handling the conditions. This allows the `AlertManager` to add a "Go To Settings" button when the user has adjusted notification permissions in an undesirable way.

File: Loop/Managers/Alerts/AlertManager.swift [hunk 1]
```diff
@@ -584,3 +584,130 @@ extension AlertManager: PresetActivationObserver {
...
+                retractAlert(identifier: alert.identifier)
+            }
+            return false
+        }
+    }
+}
+
+fileprivate extension AlertManager {
+    // MARK: Risk Mitigating Alert
+    private static let riskMitigatingAlertIdentifier = Alert.Identifier(managerIdentifier: "LoopAppManager", alertIdentifier: "riskMitigatingAlert")
```
Comment (ps2) [inline Loop/Managers/Alerts/AlertManager.swift]: "risk mitigating alert" is probably too generic.  All alerts are mitigating risks. This alert should be something like "UnsafeNotificationSettingsAlert" , or something along those lines.

File: Loop/Managers/AlertPermissionsChecker.swift [hunk 2]
```diff
@@ -81,13 +83,17 @@ public class AlertPermissionsChecker: ObservableObject {
         }
     }
 
-    func gotoSettings() {
-        UIApplication.shared.open(URL(string: UIApplication.openSettingsURLString)!)
+    static func gotoSettings() {
+        // TODO with iOS 16 this API changes to UIApplication.openNotificationSettingsURLString
```
Comment (ps2) [inline Loop/Managers/AlertPermissionsChecker.swift:87]: 👍

File: Loop/Managers/AlertPermissionsChecker.swift [hunk 3]
```diff
@@ -98,97 +104,9 @@ fileprivate extension AlertPermissionsChecker {
...
-            if alreadyIssued {
-                setAlreadyIssued(false)
-                alertManager?.retractAlert(identifier: alert.identifier)
-            }
-            return false
-        }
-    }
-}
-
-fileprivate extension AlertPermissionsChecker {
```
Comment (ps2) [inline Loop/Managers/AlertPermissionsChecker.swift:131]: Moving the definition of these particular alerts to AlertManager seems like a step backwards in terms of separation of responsibilities. 

I think it makes sense for this class to delegate to AlertManager for issuance, retraction, and general management of these alerts, but the definition/content of the alerts themselves is nice to keep close to the logic that determines the conditions.  Like we have DeviceManagers define the alert content and evaluate the conditions under which to issue/retract, but they delegate to something else for issuance/retraction.

I know the AlertManager *does* have some alert definitions, and some condition evaluation logic for Loop not Looping alerts and Bluetooth disabled alerts, but I feel the same about those; it'd probably be cleaner (and more testable) if those were split out as well.  In this case, we already have a "Checker" class split out, so it feels right to keep the alert definition and condition logic in this class. The seam between this class and the AlertManager could then just be the `AlertIssuer` protocol?

**Review by ps2 — CHANGES_REQUESTED** [review]
I think some naming needs to be clarified, and I would suggest keeping alert definitions on the class that has the responsibility for checking the condition.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 4]
```diff
@@ -835,6 +835,8 @@ final class StatusTableViewController: LoopChartsTableViewController {
             contentConfig.textProperties.adjustsFontSizeToFitWidth = true
             contentConfig.image = UIImage(systemName: "exclamationmark.triangle.fill")?.withTintColor(.red)
             contentConfig.imageProperties.tintColor = .red
+            contentConfig.secondaryText = "Enable Critical Alerts\nMore text to see how line wrapping works\nEven more text"
```
Comment (ps2) [inline Loop/View Controllers/StatusTableViewController.swift]: Looks like debug text?

File: Loop/Managers/Alerts/AlertManager.swift [hunk 5]
```diff
@@ -584,3 +584,77 @@ extension AlertManager: PresetActivationObserver {
...
+    func alertPermissions(requiresRiskMitigation: Bool, scheduledDeliveryEnabled: Bool) {
+        if !issueOrRetract(alert: AlertPermissionsChecker.unsafeNotificationPermissionsAlert,
+                           condition: requiresRiskMitigation,
+                           alreadyIssued: UserDefaults.standard.hasIssuedRiskMitigatingAlert,
+                           setAlreadyIssued: { UserDefaults.standard.hasIssuedRiskMitigatingAlert = $0 },
+                           issueHandler: { alert in
+            // the risk mitigation in-app alert is presented with a button to navigate to settings
+            self.recordIssued(alert: alert)
+            let alertController = AlertPermissionsChecker.constructUnsafeNotificationPermissionsInAppAlert() { [weak self] in
+                self?.acknowledgeAlert(identifier: AlertPermissionsChecker.unsafeNotificationPermissionsAlertIdentifier)
```
Comment (ps2) [inline Loop/Managers/Alerts/AlertManager.swift:598]: We really should be adding actions to our Alert model for this. I'm guessing you're not doing that for expediency in this case.

**Review by ps2 — APPROVED** [review]
LGTM. Just wanting to get actions added to Alerts. We've been pushing that out a long time, but it's probably not a huge lift, if we do it minimally (titles + action identifiers).

**Pull request merged**